src/masternodes/rpc_icxorderbook.cpp

```diff
@@ -0,0 +1,1479 @@
+
+
+    int targetHeight;
+    {
+        LOCK(cs_main);
+        DCT_ID idToken;
+        std::unique_ptr<CToken> token;
+
+        if (order.orderType == CICXOrder::TYPE_INTERNAL)
+        {
+            token = pcustomcsview->GetTokenGuessId(tokenFromSymbol, idToken);
+            if (!token)
+                throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Token %s does not exist!", tokenFromSymbol));
+            order.idToken = idToken;
+
+            if (!metaObj["receivePubkey"].isNull())
+                order.receivePubkey = PublickeyFromString(trim_ws(metaObj["receivePubkey"].getValStr()));
+            else
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameters, argument \"receivePubkey\" must not be null");
+
+            CTokenAmount balance = pcustomcsview->GetBalance(order.ownerAddress,idToken);
+            if (balance.nValue < order.amountFrom)
+                throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Not enough balance for Token %s on address %s!", token->CreateSymbolKey(order.idToken), ScriptToString(order.ownerAddress)));
+        }
+        else
+        {
+            token = pcustomcsview->GetTokenGuessId(tokenToSymbol, idToken);
+            if (!token)
+                throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Token %s does not exist!", tokenToSymbol));
+            order.idToken = idToken;
+        }
+
+        targetHeight = ::ChainActive().Height() + 1;
+    }
+
+    CDataStream metadata(DfTxMarker, SER_NETWORK, PROTOCOL_VERSION);
+    metadata << static_cast<unsigned char>(CustomTxType::ICXCreateOrder)
+             << order;
+
+    CScript scriptMeta;
+    scriptMeta << OP_RETURN << ToByteVector(metadata);
+
+    const auto txVersion = GetTransactionVersion(targetHeight);
+    CMutableTransaction rawTx(txVersion);
+
+    CTransactionRef optAuthTx;
+    std::set<CScript> auths{authScript};
+    rawTx.vin = GetAuthInputsSmart(pwallet, rawTx.nVersion, auths, false, optAuthTx, txInputs);
+
+    rawTx.vout.push_back(CTxOut(0, scriptMeta));
+
+    CCoinControl coinControl;
+
+    // Return change to auth address
+    CTxDestination dest;
+    ExtractDestination(*auths.cbegin(), dest);
+    if (IsValidDestination(dest))
+        coinControl.destChange = dest;
+
+    fund(rawTx, pwallet, optAuthTx,&coinControl);
+
+    // check execution
+    {
+        LOCK(cs_main);
+        CCoinsViewCache coinview(&::ChainstateActive().CoinsTip());
+        if (optAuthTx)
+            AddCoins(coinview, *optAuthTx, targetHeight);
+        const auto metadata = ToByteVector(CDataStream{SER_NETWORK, PROTOCOL_VERSION, order});
+        execTestTx(CTransaction(rawTx), targetHeight, metadata, CICXCreateOrderMessage{}, coinview);
+    }
+    UniValue ret(UniValue::VOBJ);
+    ret.pushKV("WARNING", "ICX and Atomic Swap are experimental features. You might end up losing your funds. USE IT AT YOUR OWN RISK.");
+    ret.pushKV("result",signsend(rawTx, pwallet, optAuthTx)->GetHash().GetHex());
```

Comment:
Since this is already contained in an object called "result", can you rename this property to "txid"? Same for the other RPC calls

Reply:
@Mixa84 this isn't resolved

Reply:
@monstrobishi It is resolved, please pull the latest commits.